src/modules/tests/test_data/test-data.reducer.ts

```diff
@@ -0,0 +1,31 @@
+import { TestData } from '@dvsa/mes-test-schema/categories/B';
+import * as testDataActions from './test-data.actions';
+import { createFeatureSelector } from '@ngrx/store';
+
+export const initialState: TestData = {
+  dangerousFaults: {},
+  drivingFaults: {},
+  manoeuvres: {},
+  seriousFaults: {},
+  testRequirements: {},
+};
+
+export function testDataReducer(
+  state = initialState,
+  action: testDataActions.Types,
+): TestData {
+  switch (action.type) {
+    case testDataActions.ADD_DRIVING_FAULT:
+      return {
+        ...state,
+        drivingFaults: Object.assign({}, state.drivingFaults, {
```

Comment:
You could use the destructuring operator (...state.drivingFaults) instead of `Object.assign`

Really like that we are not calculating the `newFaultCount` here.

Reply:
Made the change :)